narwhals/_spark_like/expr_dt.py

```diff
@@ -15,6 +15,29 @@ class SparkLikeExprDateTimeNamespace:
     def __init__(self, expr: SparkLikeExpr) -> None:
         self._compliant_expr = expr
 
+    def to_string(self, format: str) -> SparkLikeExpr:
+        F = self._compliant_expr._F  # noqa: N806
+
+        def _to_string(_input: Column) -> Column:
+            # Handle special formats
+            if format == "%G-W%V":
+                return self._format_iso_week(_input)
+            if format == "%G-W%V-%u":
+                return self._format_iso_week_with_day(_input)
+            if format in {"%Y-%m-%dT%H:%M:%S.%f", "%Y-%m-%dT%H:%M:%S%.f"}:
+                return self._format_iso_datetime(_input)
+
+            # Convert Python format to PySpark format
+            pyspark_fmt = strptime_to_pyspark_format(format)
+
+            result = F.date_format(_input, pyspark_fmt)
+            if "T" in format:
+                return F.replace(result, F.lit(" "), F.lit("T"))
```

Comment:
@MarcoGorelli literal "T" is not supported in pyspark from what I can see.
This is 100% a _hack_: in `strptime_to_pyspark_format` we replace "T" with " ", and here we add it back if needed.
Let me know what you think of this

Reply:
does this still work if there was a space somewhere else?

i'd be more inclined to just document the limitation

Reply:
Yep it wouldn't work. That's why I am saying it's a hack. In principle we could try to pin point where the "T" character was but honestly I don't know if it's worth it 🥲

Reply:
yeah not worth it - i think the rest looks good though

Reply:
Alright, I can fix it later yet, [this will be unresolved then](https://github.com/narwhals-dev/narwhals/pull/1842#discussion_r2105832752) since it uses a format with T